classes/reference_data.py

```diff
@@ -14,23 +16,35 @@ def __init__(self, path, yield_response=False):
         self._yield_response = yield_response
 
     def __enter__(self):
-        try:
-            response_json = requests.request(
-                "GET", self._url, headers=self._headers
-            ).json()
-
-            if self._yield_response:
-                return response_json
-            else:
-                return response_json["data"]
-
-        except Exception:
-            print("Failed to download reference data", self._url)
-            sys.exit(1)
+        return self.retry_with_backoff(self.do_fetch)
 
     def __exit__(self, _a, _b, _c):
         pass
 
+    def do_fetch(self):
+        response_json = requests.request("GET", self._url, headers=self._headers).json()
+
+        if self._yield_response:
+            return response_json
+        else:
+            return response_json["data"]
+
+    def retry_with_backoff(self, function, retries=5, backoff_in_seconds=1):
+        number_of_retries = 0
+        while True:
```

Comment:
Could this be rewritten to do a for loop on the number of retries with `sys.exit(0)` - then a `sys.exit(1)`  after the for loop? Would avoid needing an infinite loop - exit code looks fine but infinite loops are just a pattern I try to avoid where possible?

Reply:
Yeah, its definitely not usually a rubyists thing to do to use loops unless you want long lived concurrent behaviour, etc. I'll have a gander at changing it